File: 17173.py
N, M = map(int, input().split())
Mi = list(map(int, input().split()))

sum = 0
for i in range(1, N + 1):
    for j in range(M):
        if i % Mi[j] == 0:
            sum += i
            break

print(sum)

File: 17173.cpp
#include <iostream>
using namespace std;

int Mi[1000];

int main() {
	int N, M; cin >> N >> M;
	for (int i = 0; i < M; i++)
		cin >> Mi[i];

	int sum = 0;
	for (int i = 1; i <= N; i++) {
		for (int j = 0; j < M; j++) {
			if (i % Mi[j] == 0) {
				sum += i;
				break;
			}
		}
	}

	cout << sum << endl;
}
